tests/postgres_tests/test_serial.py

```diff
@@ -0,0 +1,344 @@
+                f"pg_get_serial_sequence('\"{db_table}\"','{field}'), "
+                f'coalesce(max("{field}"), 1), '
+                f'max("{field}") IS NOT null'
+                f') FROM "{db_table}";'
+            )
+
+        statements = self.get_sequence_reset_sql(SerialModel)
+        self.assertEqual(len(statements), 4)
+        self.assertEqual(statements[0], get_statement("id"))
+        self.assertEqual(statements[1], get_statement("small_serial"))
+        self.assertEqual(statements[2], get_statement("serial"))
+        self.assertEqual(statements[3], get_statement("big_serial"))
+
+    def test_reset_sequences_to_max_if_record_exists(self):
+        obj_1 = SerialModel.objects.create(small_serial=1, serial=2, big_serial=3)
+        self.assertEqual(obj_1.small_serial, 1)
+        self.assertEqual(obj_1.serial, 2)
+        self.assertEqual(obj_1.big_serial, 3)
+
+        self.reset_sequences_to_max(SerialModel)
+
+        obj_2 = SerialModel.objects.create()
+        self.assertEqual(obj_2.small_serial, 2)
+        self.assertEqual(obj_2.serial, 3)
+        self.assertEqual(obj_2.big_serial, 4)
+
+        obj_3 = SerialModel.objects.create()
+        self.assertEqual(obj_3.small_serial, 3)
+        self.assertEqual(obj_3.serial, 4)
+        self.assertEqual(obj_3.big_serial, 5)
+
+    def test_reset_sequences_to_max_if_record_doesnt_exist(self):
+        self.reset_sequences_to_max(SerialModel)
+
+        obj_1 = SerialModel.objects.create()
+        self.assertEqual(obj_1.small_serial, 1)
+        self.assertEqual(obj_1.serial, 1)
+        self.assertEqual(obj_1.big_serial, 1)
+
+        obj_2 = SerialModel.objects.create()
+        self.assertEqual(obj_2.small_serial, 2)
+        self.assertEqual(obj_2.serial, 2)
+        self.assertEqual(obj_2.big_serial, 2)
+
+    def test_bulk_create(self):
+        self.reset_sequences_to_1(SerialModel)
+        objs = [
+            SerialModel(),
+            SerialModel(small_serial=1, serial=1, big_serial=1),
+            SerialModel(serial=5),
+            SerialModel(),
+        ]
+
+        obj_1, obj_2, obj_3, obj_4 = SerialModel.objects.bulk_create(objs)
+        self.assertEqual(
+            (obj_1.small_serial, obj_1.serial, obj_1.big_serial),
+            (1, 1, 1),
+        )
+        self.assertEqual(
+            (obj_2.small_serial, obj_2.serial, obj_2.big_serial),
+            (1, 1, 1),
+        )
+        self.assertEqual(
+            (obj_3.small_serial, obj_3.serial, obj_3.big_serial),
+            (2, 5, 2),
+        )
+        self.assertEqual(
+            (obj_4.small_serial, obj_4.serial, obj_4.big_serial),
+            (3, 2, 3),
+        )
+
+
+class SerialPKModelTests(SerialModelBaseTests):
+    def test_create(self):
+        self.reset_sequences_to_1(SerialPKModel)
+
+        obj_1 = SerialPKModel.objects.create()
+        self.assertEqual(obj_1.id, 1)
+
+        obj_2 = SerialPKModel.objects.create()
+        self.assertEqual(obj_2.id, 2)
```

Comment:
```suggestion
```
I think we do basically this inside `SerialFKModelTests.test_create` so we probably don't need this